src/main/java/com/codecool/dungeoncrawl/ui/enemiesmove/GhostMove.java

```diff
@@ -0,0 +1,28 @@
+package com.codecool.dungeoncrawl.ui.enemiesmove;
+
+import com.codecool.dungeoncrawl.data.CellType;
+import com.codecool.dungeoncrawl.data.GameMap;
+
+public class GhostMove implements  EnemiesMoves{
+
+    private boolean moveUp = true;
+    private boolean moveDown;
+    @Override
+    public void perform(GameMap map, int position) {
+        if(!map.getGhost().getCell().getNeighbor(0, position + 1).getType().equals(CellType.FLOOR)){
```

Comment:
Thet is where it's getts messy. This is why I would move the `Movers` into an other layer.